bbot/modules/lightfuzz_submodules/xss.py

```diff
@@ -29,6 +29,33 @@ def determine_context(self, cookies, html, random_string):
 
         return between_tags, in_tag_attribute, in_javascript
 
+    def determine_javascript_quote_context(self, target, text):
+        # Define regex patterns for double and single quotes
+        quote_patterns = {
+            "double": f'"[^"]*{target}[^"]*"',
+            "single": f"'[^']*{target}[^']*'"
+        }
+
+        # Split the text by semicolons to isolate JavaScript statements
+        statements = text.split(";")
+
+        def is_balanced(section, target_index, quote_char):
+            left = section[:target_index]
+            right = section[target_index + len(target):]
+            return left.count(quote_char) % 2 == 0 and right.count(quote_char) % 2 == 0
+
+        for statement in statements:
+            for quote_type, pattern in quote_patterns.items():
+                match = re.search(pattern, statement)
```

Comment:
can we asyncify this regex

Reply:
https://github.com/blacklanternsecurity/bbot/pull/2173/commits/85b6ae744c57c2656955c0075ca796b4a91b5492